src/Security/MemberAuthenticator/LostPasswordHandler.php

```diff
@@ -24,108 +36,120 @@ class MemberLoginHandler extends FormRequestHandler
      * @config
      */
     private static $allowed_actions = [
-        'dologin',
-        'logout',
+        'lostpassword',
+        'LostPasswordForm',
+        'passwordsent',
     ];
 
+    private $link = null;
+
     /**
-     * Login form handler method
-     *
-     * This method is called when the user clicks on "Log in"
-     *
-     * @param array $data Submitted data
-     * @return HTTPResponse
+     * @param $link The URL to recreate this request handler
      */
-    public function dologin($data)
+    public function __construct($link)
     {
-        if ($this->performLogin($data)) {
-            return $this->logInUserAndRedirect($data);
-        }
+        $this->link = $link;
+        parent::__construct();
+    }
 
-        /** @skipUpgrade */
-        if (array_key_exists('Email', $data)) {
-            Session::set('SessionForms.MemberLoginForm.Email', $data['Email']);
-            Session::set('SessionForms.MemberLoginForm.Remember', isset($data['Remember']));
+    /**
+     * Return a link to this request handler.
+     * The link returned is supplied in the constructor
+     * @return string
+     */
+    public function link($action = null)
+    {
+        if ($action) {
+            return Controller::join_links($this->link, $action);
+        } else {
+            return $this->link;
         }
-
-        // Fail to login redirects back to form
-        return $this->redirectBackToForm();
     }
 
     /**
-     * Redirect to password recovery form
-     *
-     * @return HTTPResponse
+     * URL handler for the initial lost-password screen
      */
-    public function redirectToLostPassword()
+    public function lostpassword()
     {
-        $lostPasswordLink = Security::singleton()->Link('lostpassword');
-        return $this->redirect($this->addBackURLParam($lostPasswordLink));
+
+        $message = _t(
+            'Security.NOTERESETPASSWORD',
+            'Enter your e-mail address and we will send you a link with which you can reset your password'
+        );
+
+        return [
+            'Content' => DBField::create_field('HTMLFragment', "<p>$message</p>"),
```

Comment:
The other option is using a separate template, and wrap the message in a <p> tag there, which is not something I would even consider recommending.